libs/zephyr-agent/src/lib/node-persist/app-deploy-result-cache.ts

```diff
@@ -1,25 +1,53 @@
-import { getItem, removeItem, setItem } from 'node-persist';
+import { forEach, getItem, keys, removeItem, setItem } from 'node-persist';
+import type { Snapshot } from 'zephyr-edge-contract';
 import { storage } from './storage';
 import { StorageKeys } from './storage-keys';
 
+const prefix = `${StorageKeys.ze_app_deploy_result}:`;
+
+export interface DeployResult {
+  urls: string[];
+  snapshot: Snapshot;
+}
+
 export async function setAppDeployResult(
   application_uid: string,
-  value: { urls: string[] }
+  value: DeployResult
 ): Promise<void> {
   await storage;
-  void (await setItem(`${StorageKeys.ze_app_deploy_result}:${application_uid}`, value, {
+  void (await setItem(`${prefix}${application_uid}`, value, {
     ttl: 1000 * 60 * 60 * 24,
   }));
 }
 
 export async function getAppDeployResult(
   application_uid: string
-): Promise<{ urls: string[] } | undefined> {
+): Promise<DeployResult | undefined> {
   await storage;
-  return getItem(`${StorageKeys.ze_app_deploy_result}:${application_uid}`);
+  return getItem(`${prefix}${application_uid}`);
 }
 
 export async function removeAppDeployResult(application_uid: string): Promise<void> {
   await storage;
-  await removeItem(`${StorageKeys.ze_app_deploy_result}:${application_uid}`);
+  await removeItem(`${prefix}${application_uid}`);
+}
+
+export async function getAllDeployedApps(): Promise<string[]> {
+  const allKeys = await keys();
+  const resultKeys = allKeys.filter((key) => key.startsWith(prefix));
```

Comment:
we discuss on a call to have a catalog entry which will contain a registry of apps a HashMap (Record<T,K>)